### ISSUE TITLE
Target output rendering suffers glitches if Synfig Studio has to render its buffer

### ISSUE THREAD

**Issue by FirasH** [synfig/synfig#1455]
**Synfig version**:
master 4c519a9

**Issue description**:
1. Open an animation and render it to a target output (tested on .mp4 and .png sequence)
2. While rendering, change `Current time` by clicking on different positions on the `Timebar`
More changes result in more glitches (one per change?)

The issue is quite relevant, as the user is getting a corrupted target output and, at least in my case, to get a clean output it's need to leave Synfig Studio process the output without e.g. checking the animation. (btw changing layers while rendering changes also the target)

At the following [link](https://drive.google.com/drive/folders/1H1KHFwfcv0aLnGhjY_ZqmBXmRgWDvhQ_?usp=sharing) you can find the:

- **animation project**: original animation file on which I detected this issue
- **animation export**: target output, includes both the corrupted .png sequence, and three test .mp4 files with self explanatory names

**Comment by FirasH** [comment]
The link was not accessible, now it's fixed.

I've also reproduced this issue on another machine.

**Comment by rodolforg** [comment]
I don't know why changing the current time corrupts the rendering output, but I can see why layer changing does:
https://github.com/synfig/synfig/blob/894386769881efe2702ff18b71054e6b1e9224fa/synfig-studio/src/gui/render.cpp#L407
It just get the _current_ canvas _handle_.
It should clone or serialize/deserialize the current canvas.

**Comment by morevnaproject** [comment]
Exactly. Rendering operation should create a copy of opened canvas, open it (internally) and render.

Maybe we can utilize autobackup infrastructure for that? It is already used to create copies when we call plugins...
